### PR TITLE
feat: introduce new help overlay for add section button

### PR DESCRIPTION
**Pull request by dcshzj**
## Problem

The add section button does not have the ability to show a help overlay.

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
- [x] No - this PR is backwards compatible

**Features**:

- The add section button now has a new help overlay component, which allows a popover to appear on the right side of the option upon hover.
- The Editable component is also promoted to a shared component, and stories were added for the AddSectionButton to showcase the new help overlay.

## Before & After Screenshots

**BEFORE**:

Screenshot 2023-09-21 at 11 39 27 AM

**AFTER**:

Screenshot 2023-09-21 at 11 59 28 AM

## Tests

- [ ] Unit tests (using `npm run tests`)
- [ ] e2e tests (comment on this PR with the text `!run e2e`)
- [x] Smoke tests
    - [ ] Run the storybook and navigate to components/AddSectionButton
    - [ ] Ensure that the help overlay appears for text cards, info-columns and announcements
    - [ ] Navigate to any site
    - [ ] Ensure that there are no regressions in the Editable functionality (dnd, accordion, etc) for homepage, contact us and navigation bar pages

## Deploy Notes

*None*

### CODE REVIEW

**Comment by dcshzj** [comment]
!run chromatic

**Comment by dcshzj** [comment]
!run chromatic

**Comment by dcshzj** [comment]
@sehyunidaaa Chromatic review is here: https://www.chromatic.com/review?appId=61fcc8e14c3af9003a102ac9&number=1515&type=linked&view=changes

**Review by kishore03109 — APPROVED** [review]
Approving since my comments are non-functional. 
I do think think the storybook code can be made a lot neater for extensibility in the future

Biz logic wise lgtm

File: src/components/Editable/AddSectionButton.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+} from "assets/images"
+
+import { AddSectionButton } from "./AddSectionButton"
+
+const addSectionButtonMeta = {
+  title: "Components/AddSectionButton",
+  component: AddSectionButton,
+} as Meta<typeof AddSectionButton>
+
+interface AddSectionButtonTemplateArgs {
```
[the remark this answers is not in the document]
Comment (dcshzj) [reply src/components/Editable/AddSectionButton.stories.tsx:24]: Adjusted in b688a481aa10e583321265881a067dd6c7c0c5c6! I think I got a little carried away here oops.

**Pull request merged**